src/eos-updater-util.c

```diff
@@ -313,73 +323,180 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
       g_set_error (error, G_IO_ERROR, G_IO_ERROR_FAILED,
                    "Failed to download branch config data (HTTP %d),"
                    " cannot upgrade", status);
-      goto out;
+      return NULL;
     }
 
   bkf = g_key_file_new ();
   if (!g_key_file_load_from_data (bkf, msg->response_body->data, -1,
                                   G_KEY_FILE_NONE, error))
-    goto out;
+    return NULL;
+  return g_steal_pointer (&bkf);
+}
+
+static gboolean
+process_single_group (GKeyFile *bkf,
+                      const gchar *group_name,
+                      gboolean *on_hold,
+                      gchar **p_ref,
+                      GError **error)
+{
+  if (g_key_file_get_boolean (bkf, group_name, ON_HOLD_KEY, NULL))
```

Comment:
Hm, looks like we have to handle it a bit more specially - the OnHold key might be missing and that's fine, we don't want to propagate the error in this case, right?